sonar-dotnet-shared-library/src/main/java/org/sonar/plugins/dotnet/tests/VisualStudioCoverageXmlReportParser.java

```diff
@@ -119,27 +133,51 @@ private void handleSourceFileTag(XmlParserHelper xmlParserHelper) {
         return;
       }
 
-      if (coveredLines.containsKey(id)) {
-        LOG.trace("Found covered lines for id '{}' for path '{}'", id, canonicalPath);
-
-        for (Integer line : coveredLines.get(id)) {
-          coverage.addHits(canonicalPath, line, 1);
-        }
+      Map<Integer, List<Boolean>> fileCoverage = coveragePerLinePerFile.get(id);
+      if (!fileCoverage.isEmpty())
+      {
+        LOG.trace("Found coverage information about '{}' lines for file id '{}' , path '{}'",
+          fileCoverage.size(), id, canonicalPath);
       }
```

Comment:
Since the trace log is about line coverage we could move the log in `processLineCoverage`